community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -138,7 +138,7 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
 
   override def notifications = Iterable.empty[InternalNotification]
 
-  override def hasNext = innerIterator.hasNext
+  override def hasNext = !taskCloser.isClosed && innerIterator.hasNext
```

Comment:
Is the contract of `hasNext` such that calling it when a previous call returned `false` produces undefined behaviour? If not, we need this kind of check at some point between calling `hasNext` and entering `accept` method.